lib/buffer.js

```diff
@@ -1001,11 +1011,22 @@ function _fill(buf, value, offset, end, encoding) {
       return buf;
   }
 
-  const res = bindingFill(buf, value, offset, end, encoding);
-  if (res < 0) {
-    if (res === -1)
-      throw new ERR_INVALID_ARG_VALUE('value', value);
-    throw new ERR_BUFFER_OUT_OF_BOUNDS();
+
+  if (typeof value === 'number') {
+    // OOB check
+    const byteLen = TypedArrayProto_byteLength.call(buf);
+    const fillLength = end - offset;
+    if (offset > end || fillLength + offset > byteLen)
+      throw new ERR_BUFFER_OUT_OF_BOUNDS();
+
+    TypedArrayFill.call(buf, value, offset, end);
+  } else {
+    const res = bindingFill(buf, value, offset, end, encoding);
+    if (res < 0) {
+      if (res === -1)
+        throw new ERR_INVALID_ARG_VALUE('value', value);
+      throw new ERR_BUFFER_OUT_OF_BOUNDS();
```

Comment:
Future enhancement: `res < 0` is sometimes a coercion-to-nan, sometimes not because `bindingFill()` returns either `undefined` or a number.

Changing `Fill()` in node_buffer.cc to always return a number is probably beneficial for performance.